threatmodels/notation-theatmodel.md

```diff
@@ -0,0 +1,84 @@
+- **Artifact Builder**: The actor who produces software artifacts in a build system. The software artifacts to be signed are arbitrary artifacts, which could be container images, helm charts in the context of Cloud, SBOM (Software Bill of Materials) in the context of secure supply chain, and other artifacts. Normally the artifacts are produced by the build process with vulnerabilities scanned, analyzed and fixed. Notation supports signing and verifying artifacts stored in registries or in the file system as OCI image layout.
+  - Artifact in registries: Take container images as an example, users can use `docker` commands to produce container images, and store them to a registry. A container image in a registry can be referenced by a fully qualified path, for example `registry.wabbit-networks.io/net-monitor@sha256:xxx`. SBOM is another example, which are generated by tools during build process, stored as OCI artifacts in OCI-compliant registries, and referenced by a fully qualified path as well as container images.
+  - Artifact as OCI image layout: OCI image layout is a directory structure that contains files and folders that refer to an OCI image. See [OCI image layout spec](https://github.com/opencontainers/image-spec/blob/v1.0.0/image-layout.md) for details.
+- **File System**: The file system where Notation or application built with Notation library is installed. For Notation signing artifacts as OCI image layout, it is the place where artifacts and associated signatures are stored. There are different directories and files as follows:
+  - Notation config file, where general configuration is stored
+  - Notation trust policy file, where trust policy configuration is stored
+  - Notation trust store, where certificate files are stored to validate signatures
+  - Notation plugin directory, where Notation plugins are installed
+  - Credential store, where the registry or Key management credentials are stored
+- **Notation Plugin**: An external component built by different vendors, and integrated to Notation as a signing plugin for signature generation, or as a verifying plugin for signature verification, see [plugin spec](https://github.com/notaryproject/notaryproject/blob/v1.0.0-rc.2/specs/plugin-extensibility.md) for details.
+- **Registry**: A registry for Docker and Open Container Initiative (OCI) images, with support for all OCI artifacts. It is the storage where artifacts are pushed to by artifact builder. Registries are outside Notation trust boundary.
+- **KMS**: A remote Key Management Service provides key and associated certificate management, and remote signing service. KMSs are outside Notation trust boundary.
+- **OCSP Responder**: The [Online Certificate Status Protocol (OCSP)](https://en.wikipedia.org/wiki/Online_Certificate_Status_Protocol) is an Internet protocol used for obtaining the revocation status of an X.509 digital certificate. Notation uses OCSP to validate certificate revocation status from OCSP Responder.
```

Comment:
Reference RFC doc (i.e. [RFC 2560](https://www.rfc-editor.org/rfc/rfc2560)) instead of Wikipedia since Wikipedia cannot be used as an academic knowledge source.